project/src/com/google/daggerquery/executor/QueryExecutor.java

```diff
@@ -67,27 +72,22 @@ public static void main(String[] args) {
     ImmutableList.Builder<String> resultBuilder = new ImmutableList.Builder();
 
     // We assume that we successfully executed a query only if in at least one graph it was executed without fail.
-    MisspelledNodeNameException lastMisspelledNodeNameException = null;
-    IllegalArgumentException lastIllegalArgumentException = null;
+    Set<Entry<Integer, Exception>> exceptions = new TreeSet<>(Entry.comparingByKey());
```

Comment:
`ImmutableSetMultimap` is your friend. Example usage: https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableSetMultimap.Builder.html

Reply:
😼😼😼
Thanks, [done](https://github.com/googleinterns/dagger-query/pull/44/commits/b8253b2e085e45ecdb327ad7fbe689d75f0a084a)!